File: app/Models/StockLocation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Traits\SalonOwned;

class StockLocation extends Model 
{
    use SalonOwned;
    
    protected $fillable = [
        'salon_id',
        'name',
        'is_default',
        'meta'
    ];
    
    protected $casts = [
        'is_default' => 'boolean',
        'meta' => 'array'
    ];
    
    public function stockItems()
    {
        return $this->hasMany(StockItem::class, 'location_id');
    }
    
    public function stockMovements()
    {
        return $this->hasMany(StockMovement::class, 'location_id');
    }
}

File: salonmanager/frontend/lib/features/inventory/models/stock_location.dart
class StockLocation {
  final int id;
  final String name;
  final bool isDefault;
  final Map<String, dynamic>? meta;

  StockLocation({
    required this.id,
    required this.name,
    required this.isDefault,
    this.meta,
  });

  factory StockLocation.fromJson(Map<String, dynamic> json) => StockLocation(
    id: json['id'],
    name: json['name'],
    isDefault: json['is_default'] ?? false,
    meta: json['meta'] != null ? Map<String, dynamic>.from(json['meta']) : null,
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'is_default': isDefault,
    'meta': meta,
  };
}